IPython/core/magics/basic.py

```diff
@@ -454,7 +454,7 @@ def doctest_mode(self, parameter_s=''):
         save_dstore('rc_active_types',disp_formatter.active_types)
         save_dstore('prompt_templates',(pm.in_template, pm.in2_template, pm.out_template))
 
-        if mode == False:
+        if not mode:
```

Comment:
can mode be None? Doesn't seem like it should be. I don't know what it would mean if it were.